Spigot-API-Patches/0167-Add-BaseComponent-array-to-a-variety-of-events.patch

```diff
@@ -0,0 +1,2079 @@
++                switch (current) {
++                    /* Flag characters */
++                    case '<':
++                    case '-':
++                    case 'e':
++                    case 'f':
++                    case 'i':
++                    case 'E':
++                    case 'F':
++                    case 'I':
++
++                    /* parameter values */
++                    case '.':
++                    case '$':
++                        continue conversion_search;
++
++                    /* Digits */
++                    default:
++                        if (current >= '0' && current <= '9') {
++                            continue conversion_search;
++                        }
++                }
++
++                /* Invalid */
++                return -1;
++            }
++
++            if (conversion == 0) {
++                /* No conversion found */
++                return -1;
++            }
++
++            /* Parameters are read BACKWARDS as this simplifies the logic involved */
++            /* currentOffset however may never be less-than off, but it may be equal to */
++            int currentOffset = conversionIndex - 1; /* start at character before conversion */
++
++            int width; /* will default to 0 */
++            int precision = Integer.MAX_VALUE;
++
++            boolean readPrecision = false;
++            /* This loop should read both the width and precision if they exist */
++            /* Otherwise they will remain at the default value */
++
++            /* Note that this loop will break immediately if there is no width or precision */
++            /* As the loop only continues if the precision is read */
++            for (;;) {
++                int currentNumber = 0;
++                for (int multiplier = 1; currentOffset > off; --currentOffset, multiplier *= 10) {
++                    final char current = string.charAt(currentOffset);
++                    if (current < '0' || current > '9') {
++                        break;
++                    }
++                    currentNumber += multiplier * (current - '0');
++                }
++                if (string.charAt(currentOffset) == '.') {
++                    /* Read the precision */
++                    if (readPrecision) {
++                        /* Precision may not be specified twice */
++                        return -1;
++                    }
++                    readPrecision = true;
++
++                    precision = currentNumber;
++                } else {
++                    /* Read the width */
++                    width = currentNumber;
++                    /* Next to read is the flags */
++                    break;
++                }
++            }
++
++            /* Next up to read is the flags value */
++
++            int flags = 0;
++
++            for (char current = string.charAt(currentOffset);current != '$' && current != '%'; current = string.charAt(--currentOffset)) {
```

Comment:
indent condition by 1 space

Reply:
also note that the current character can only be $, '%', or a flag at this stage